core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1744,34 +1744,6 @@ private[spark] object Utils extends Logging {
     hashAbs
   }
 
-  /**
-   * NaN-safe version of `java.lang.Double.compare()` which allows NaN values to be compared
-   * according to semantics where NaN == NaN and NaN is greater than any non-NaN double.
-   */
-  def nanSafeCompareDoubles(x: Double, y: Double): Int = {
-    val xIsNan: Boolean = java.lang.Double.isNaN(x)
-    val yIsNan: Boolean = java.lang.Double.isNaN(y)
-    if ((xIsNan && yIsNan) || (x == y)) 0
-    else if (xIsNan) 1
-    else if (yIsNan) -1
-    else if (x > y) 1
-    else -1
-  }
-
-  /**
-   * NaN-safe version of `java.lang.Float.compare()` which allows NaN values to be compared
-   * according to semantics where NaN == NaN and NaN is greater than any non-NaN float.
-   */
```

Comment:
Do know much about the context, but `java.lang.Float.compare()` do follow the same NaN-safe behavior:

```scala
scala> java.lang.Float.compare(java.lang.Float.NaN, java.lang.Float.NaN)
res11: Int = 0
```

so this remove looks good.

Reply:
Yep, I checked with a little test:

```
val values = Seq(Double.NaN, 0, Double.MinValue, Double.MaxValue, Double.PositiveInfinity, Double.NegativeInfinity)
for (a <- values; b <- values) {
  require(java.lang.Double.compare(a, b) == Utils.nanSafeCompareDoubles(a, b))
}
```

Reply:
Yes. I also checked that `compare` handles different NaN values correctly via `floatToIntBits`, too. (Refer #23851)
```
    public static int compare(float f1, float f2) {
        if (f1 < f2)
            return -1;           // Neither val is NaN, thisVal is smaller
        if (f1 > f2)
            return 1;            // Neither val is NaN, thisVal is larger

        // Cannot use floatToRawIntBits because of possibility of NaNs.
        int thisBits    = Float.floatToIntBits(f1);
        int anotherBits = Float.floatToIntBits(f2);

        return (thisBits == anotherBits ?  0 : // Values are equal
                (thisBits < anotherBits ? -1 : // (-0.0, 0.0) or (!NaN, NaN)
                 1));                          // (0.0, -0.0) or (NaN, !NaN)
    }
```